File: exercise13.py
# Exercise 13 - Probability
# Edited by Tony Xiang
# Last Modified in 2018-7-17

import random

N = 45
total = 1000
MAX = 100000
group = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
prob = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
for i in range(1, total):
	r = random.randint(1, MAX)
	a = pow(N, r)
	b = str(a)
	c = int(b[0])
	group[c] += 1

for i in range(0, 10):
	prob[i] = group[i] / sum(group) 
print(prob) 

File: Exercises/exercise13.m
% Exercise 13 - Probability
% Edited by Tony Xiang
% Last Modified in 2018-7-17

N = 9;
total = 50;
G = zeros(9, 1);
for n = 1:total
    a = N ^ n;
    b = num2str(a);
    c = int32(str2double(b(1)));
    G(c) = G(c) + 1; 
end

P = G / sum(G)

% p1,30% > p2,17% > p3,12% > p4,9% > p5,8% > p6,7% > p7,6% > p8,5% > p9,4%